lib/livebook/session/data.ex

```diff
@@ -58,7 +60,8 @@ defmodule Livebook.Session.Data do
           secrets: %{(name :: String.t()) => value :: String.t()},
           mode: session_mode(),
           apps: list(app()),
-          app_data: nil | app_data()
+          app_data: nil | app_data(),
+          hub: nil | Provider.t()
```

Comment:
Any reason we wouldn't default to the personal hub? Otherwise we need to handle the `nil` path whenever we refer to the hub.

Reply:
I'm basing on the task from the issue:
`Before the Notebook title, adds a rounded button with the Hub's emoji (or with a placeholder to select the Hub)`

I'm okay with both. Your call

Reply:
I see! Since many features rely on the hub (currently secrets and apps), we would need to account for no-hub in the corresponding UIs and it may also be confusing that they can't use those features right away, so I think a default is justified. @josevalim?

Reply:
Defaulting to the personal Hub is a great idea. :+1: